### ISSUE TITLE
a new build_runner is available

### ISSUE THREAD

**Issue by zoechi** [davidmarne/built_redux#56]
but not yet supported by built_redux

**Comment by zoechi** [comment]
and a new `built`

**Comment by davidmarne** [comment]
Updating would be tricky before https://github.com/google/built_value.dart/pull/342 is merged, as built_redux has a dev_dependency on built_value_generator. I'm gonna wait until that gets merged to proceed with built_redux.

Also, the environment restrictions imposed by the new version of build_runner is a concern to me.
`sdk: '>=2.0.0-dev.20 <2.0.0'`

Given the fact that there is no confirmed release date for dart 2, I might move forward with a pre-release, rather than a normal release. Thoughts?

**Comment by zoechi** [comment]
Looks like the built_value PR is close to being merged.

I think moving forward with a pre-release is fine.

**Comment by zoechi** [comment]
The new built_value should be available

**Comment by davidmarne** [comment]
So i took a look at this. I created a build.yaml, however when I try to generate files I get:
```
DuplicateAssetNodeError: GeneratedAssetNode: built_redux|lib/builder.g.dart generated from input built_redux|lib/builder.dart.
This probably means you have multiple actions that are trying to output the same file
```
Which makes sense, since both are using source_gen's PartBuilder.

I can get it working if I generate with a different extension than '.g.dart', however that is obviously not ideal.

**Comment by davidmarne** [comment]
That might have to be the solution, I really don't see a way around it.

**Comment by zoechi** [comment]
You could ask the developers of the build package. I didn't have a chance yet to have a closer look at the new ways

**Comment by davidmarne** [comment]
create https://github.com/dart-lang/build/issues/1030

**Comment by davidmarne** [comment]
looks like there is already an issue here: https://github.com/dart-lang/build/issues/975

**Comment by btastic** [comment]
So there is no way to use your library right now? I was excited to try this after reading on Brian Egans repository. But all I get is errors. 

Running the generator script is giving me
```dart
'package:mime/src/mime_multipart_transformer.dart': malformed type: line 33 pos 13: cannot resolve class 'StreamTransformerBase' from 'MimeMultipartTransformer'
    extends StreamTransformerBase<List<int>, MimeMultipart> {
```

**Comment by davidmarne** [comment]
@btastic just in case you didn't see, I did comment on your flutter_built_redux issue with a couple potential workarounds to that error, which should get you going using build_runner 0.6.x

However, it still looks like there are some issues being worked out over at github.com/dart-lang/build that I would like to see get resolved before committing to supporting build_runner 0.7.x in built_redux

**Comment by zoechi** [comment]
Is this still blocked?

**Comment by davidmarne** [comment]
@zoechi sorry to leave u hanging :(

It won't be perfect until https://github.com/dart-lang/source_gen/issues/319 is resolved, but there aren't any blockers for pushing out a dev branch. I just published a dev tag, 7.4.0-dev, to pub. You can see the branch and instructions for configuring build_runner to work with built_value and built_redux here:

https://github.com/davidmarne/built_redux/tree/7.4.0-dev

**Comment by zoechi** [comment]
@davidmarne no worries
Thanks a lot for your great work :)
